src/coreclr/jit/emitriscv64.cpp

```diff
@@ -4961,13 +4967,11 @@ regNumber emitter::emitInsTernary(instruction ins, emitAttr attr, GenTree* dst,
                 {
                     if (src1->gtType == TYP_INT)
                     {
-                        emitIns_R_R_I(INS_slli, EA_8BYTE, regOp1, regOp1, 32);
-                        emitIns_R_R_I(INS_srli, EA_8BYTE, regOp1, regOp1, 32);
+                        emitIns_R_R(INS_sext_w, EA_8BYTE, src1Reg, src1Reg);
```

Comment:
Did you test your codes in coreclr and corefx? I think this code is totally wrong.

Reply:
Sorry I somehow misinterpreted the type of the shift. Revoked this changes. I will mark PR as ready after exhaustive testing